src/fragments/lib/auth/common/mfa/flows.mdx

```diff
@@ -0,0 +1,206 @@
+The Auth category supports Multi-factor Authentication (MFA) for user sign-in flows. MFA is an extra layer of security used to make sure that people trying to gain access to an online account are who they say they are. It requires users to provide additional information to verify their identity. The category supports the following MFA methods:
+* SMS
+* TOTP (Time-based One-time Password)
+
+## Set Up Backend Resources
+
+<Callout warning>
+
+When using Amplify CLI to set up backend resources, the following options are only available when starting a new project (via `amplify add auth`). You will not have access to these settings after creation (via `amplify update`).
+
+- Required MFA (i.e. Setting MFA to "ON")
+
+</Callout>
+
+<Callout>
+Note: If you create or update an SMS MFA configuration for your Cognito user pool, the Cognito service will send a test SMS message to an internal number in order to verify your configuration. You will be charged for these test messages by Amazon SNS. 
+
+Refer [SMS sandbox](/lib/auth/sms_flows#sandbox-mode).
+
+For information about Amazon SNS pricing, see [Worldwide SMS Pricing](https://aws.amazon.com/sns/sms-pricing/). 
+</Callout>
+
+<BlockSwitcher>
+
+<Block name="New Project">
+
+Run `amplify add auth` to create a new Cognito Auth resource, and follow the prompts below depending on how you want to integrate MFA into your flow.
+
+Turning MFA "ON" will make it required for all users, while "Optional" will make it available to enable on a per-user basis.
+
+import all2 from "/src/fragments/lib/auth/common/mfa/add_mfa.mdx";
+
+<Fragments fragments={{all: all2}} />
+
+</Block>
+
+<Block name="Existing Auth">
+
+Run `amplify update auth` and follow the prompts as guided below.
+
+import all5 from "/src/fragments/lib/auth/common/mfa/update_mfa.mdx";
+
+<Fragments fragments={{all: all5}} />
+
+</Block>
+
+</BlockSwitcher>
+
+## Multi-factor authentication with SMS
+
+<Callout>
+
+<b>AWS Cognito Limitation</b>
+
+The `fetchMFAPreference` API will only return enabled and preferred MFA methods, if the preferences were set using the `updateMFAPreference` API. If the preferences were not set using the `updateMFAPreference` API, the MFA preferences will return `nil` for enabled and preferred MFA methods. 
+
+Therefore it is recommended to use the `updateMFAPreference` API to set the MFA preferences for the user, if the user decides to set up any MFA methods. An example use case would be that when a user successfully sets up TOTP using the `setupTOTP` and `verifyTOTPSetup` API, `updateMFAPreference` should be invoked to set the MFA preferences for the user i.e. marking TOTP as enabled.
+
+</Callout>
+
+### Adding phone number during Sign Up
+
+Sign up users normally with the chosen `Username` type and password. `"phone_number"` will be required in the `userAttributes` if SMS MFA is ON, or manually enabled for the user
```

Comment:
There's some nuance here, though. If the username attribute is `phone_number` then it must not be included in `userAttributes` IIRC

Reply:
Got it. Updated the wording to include a note.